care/facility/api/viewsets/asset.py

```diff
@@ -118,6 +119,15 @@ def get_facility(self):
     def perform_create(self, serializer):
         serializer.save(facility=self.get_facility())
 
+    def destroy(self, request, *args, **kwargs):
+        instance = self.get_object()
+        if instance.bed_set.count():
+            raise ValidationError("Cannot delete a Location with associated Beds")
+        if instance.asset_set.count():
+            raise ValidationError("Cannot delete a Location with associated Assets")
```

Comment:
Since we have soft deletes, do reverse relations queryset account for it? Shouldn't we filter by `deleted=False`?

cc: @sainak 

```suggestion
        if instance.bed_set.filter(deleted=False).count():
            raise ValidationError("Cannot delete a Location with associated Beds")
        if instance.asset_set.filter(deleted=False).count():
            raise ValidationError("Cannot delete a Location with associated Assets")
```